pkg/gardenlet/controller/shoot/shoot/reconciler_reconcile.go

```diff
@@ -271,12 +271,18 @@ func (r *Reconciler) runReconcileShootFlow(ctx context.Context, o *operation.Ope
 			Fn:           flow.TaskFn(botanist.DeployEtcd).RetryUntilTimeout(defaultInterval, helper.GetEtcdDeployTimeout(o.Shoot, defaultTimeout)),
 			Dependencies: flow.NewTaskIDs(initializeSecretsManagement, deployCloudProviderSecret, waitUntilBackupEntryInGardenReconciled, waitUntilEtcdBackupsCopied),
 		})
-		_ = g.Add(flow.Task{
+		destroySourceBackupEntry = g.Add(flow.Task{
 			Name:         "Destroying source backup entry",
 			Fn:           botanist.DestroySourceBackupEntry,
-			SkipIf:       !allowBackup,
+			SkipIf:       !allowBackup || botanist.Seed.GetInfo().Spec.Backup == nil || !botanist.IsRestorePhase(),
```

Comment:
`!allowBackup` and `botanist.Seed.GetInfo().Spec.Backup == nil` are the same thing ref: https://github.com/gardener/gardener/blob/3da651ee4b326541b8270ab06cf421d3a85743d4/pkg/gardenlet/controller/shoot/shoot/reconciler_reconcile.go#L106

Please use `!allowBackup`